docs/source/user-guide/concepts/expressions-and-contexts.md

```diff
@@ -0,0 +1,192 @@
+Here, we group by a combination of decade of birth and whether the person is shorter than 1.7 metres:
+
+{{code_block('user-guide/concepts/expressions','group_by-2',['group_by'])}}
+
+```python exec="on" result="text" session="user-guide/concepts/expressions-and-contexts"
+--8<-- "python/user-guide/concepts/expressions.py:group_by-2"
+```
+
+The resulting dataframe, after applying aggregating expressions, contains one column per each grouping expression on the left and then as many columns as needed to represent the results of the aggregating expressions.
+In turn, we can specify as many aggregating expressions as we want:
+
+{{code_block('user-guide/concepts/expressions','group_by-3',['group_by'])}}
+
+```python exec="on" result="text" session="user-guide/concepts/expressions-and-contexts"
+--8<-- "python/user-guide/concepts/expressions.py:group_by-3"
+```
+
+See also `group_by_dynamic` and `group_by_rolling` for other grouping contexts.
+
+## Expression expansion
+
+The last example contained two grouping expressions and three aggregating expressions, and yet the resulting dataframe contained six columns instead of five.
+If we look closely, the last aggregating expression mentioned two different columns: “weight” and “height”.
+
+Polars expressions support a feature called _expression expansion_.
+Expression expansion is like a shorthand notation for when you want to apply the same transform to multiple columns.
+As we have seen, the expression
+
+```python
+pl.col("weight", "height").mean().name.prefix("avg_")
+```
+
+will compute the mean value of the columns “weight” and “height” and will rename them as “avg_weight” and “avg_height”, respectively.
+
+In this case, this expression expands into two independent expressions that Polars can execute in parallel.
+In other cases, we may not be able to know in advance how many independent expressions an expression will unfold into.
+
+Consider this simple but elucidative example:
+
+```python
+(pl.col(pl.Float64) * 1.1).name.suffix("*1.1")
+```
+
+This expression will multiply all columns with data type `Float64` by `1.1`.
+The number of columns this applies to depends on the schema of each dataframe.
+In the case of the dataframe we have been using, it applies to two columns:
+
+{{code_block('user-guide/concepts/expressions','expression-expansion-1',['group_by'])}}
+
+```python exec="on" result="text" session="user-guide/concepts/expressions-and-contexts"
+--8<-- "python/user-guide/concepts/expressions.py:expression-expansion-1"
+```
+
+In the case of the dataframe `df2` below, the same expression expands to 0 columns:
+
+{{code_block('user-guide/concepts/expressions','expression-expansion-2',['group_by'])}}
+
+```python exec="on" result="text" session="user-guide/concepts/expressions-and-contexts"
+--8<-- "python/user-guide/concepts/expressions.py:expression-expansion-2"
+```
+
+It is equally easy to imagine a scenario where the same expression would expand to dozens of columns.
+
```

Comment:
We can inspect how an expression will evaluate against a given schema:

```python
schema = pl.Schema({
    "int_1": pl.Int16,
    "int_2": pl.Int32,
    "float_1": pl.Float64,
    "float_2": pl.Float64,
    "float_3": pl.Float64,
})

print(
    pl.LazyFrame(schema=schema).select(
  (pl.col(pl.Float64) * 1.1).name.suffix("*1.1") 
).explain()
)
```
```
 SELECT [[(col("float_1")) * (1.1)].alias("float_1*1.1"), [(col("float_2")) * (1.1)].alias("float_2*1.1"), [(col("float_3")) * (1.1)].alias("float_3*1.1")] FROM
  DF ["int_1", "int_2", "float_1", "float_2"]; PROJECT 3/5 COLUMNS; SELECTION: None

```

Doesn't have to be included. Up to you.